system/lib/compiler-rt/lib/sanitizer_common/sanitizer_emscripten.cc

```diff
@@ -85,21 +87,10 @@ uptr internal_munmap(void *addr, uptr length) {
   return emscripten_builtin_munmap(addr, length);
 }
 
-extern "C" {
-  extern char __data_end;
-  extern char __heap_base;
-}
-
-inline uptr get_stack_bottom() {
-  // Align to 8 byte
-  return ((uptr) &__data_end + 7) & ~7;
-}
-
-
 void GetThreadStackTopAndBottom(bool at_initialization, uptr *stack_top,
                                 uptr *stack_bottom) {
-  *stack_top = (uptr) &__heap_base;
-  *stack_bottom = get_stack_bottom();
+  *stack_top = EM_ASM_INT({ return STACK_BASE; });
+  *stack_bottom = EM_ASM_INT({ return STACK_MAX; });
```

Comment:
Is this just more precise than what was there previously?

Reply:
Yes, and it will also work with other threads.